cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceClassLoader.java

```diff
@@ -232,26 +234,21 @@ private static PluginInstantiator createArtifactPluginInstantiator(MapReduceCont
      *
      * Plugin Lib ClassLoader, Plugin Export-Package ClassLoader, ...
      */
-    private static List<ClassLoader> createFilteredPluginClassLoaders(Configuration hConf,
-                                                                      Map<String, Plugin> plugins,
-                                                                      @Nullable PluginInstantiator pluginInstantiator) {
-      if (pluginInstantiator == null) {
+    private static List<ClassLoader> createFilteredPluginClassLoaders(String namespace, Map<String, Plugin> plugins,
+                                                                      Map<String, String> artifactFileNames,
+                                                                      PluginInstantiator pluginInstantiator) {
+      if (plugins.isEmpty()) {
         return ImmutableList.of();
       }
 
       try {
         Multimap<Plugin, String> artifactPluginClasses = getArtifactPluginClasses(plugins);
-        LocationFactory locationFactory;
-        LocationFactory localLocationFactory = new LocalLocationFactory();
-        LocationFactory hdfsLocationFactory = new HDFSLocationFactory(hConf);
-
-        // Need appropriate LocationFactory since we only the Location URI from Plugin
-        locationFactory = (MapReduceTaskContextProvider.isLocal(hConf)) ? localLocationFactory : hdfsLocationFactory;
         List<ClassLoader> pluginClassLoaders = Lists.newArrayList();
-        for (Plugin plugin : plugins.values()) {
-          ArtifactDescriptor artifactDescriptor = new ArtifactDescriptor(plugin.getArtifactId(),
-            locationFactory.create(
-              plugin.getLocationURI()));
+        for (Map.Entry<String, Plugin> pluginEntry : plugins.entrySet()) {
```

Comment:
the old `for (Plugin plugin : plugins.values())` is more succinct since you aren't using the key